eclass/kernel-build.eclass

```diff
@@ -20,6 +20,14 @@
 # the kernel and installing it along with its modules and subset
 # of sources needed to build external modules.
 
+# @ECLASS_VARIABLE: KV_FULL
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# A string containing the full kernel release version, e.g.
+# '6.9.6-gentoo-dist'. This is used to ensure consistency between the
+# kernel's release version and Gentoo's tooling. This is set by
+# kernel-build_src_configure() once we have a kernel.release file.
```

Comment:
Maybe make it explicit that if it's set, it must match the file, and if it's unset, it will be set to the value from the file. Ideally, make it clear when it needs to be set too.

Reply:
What is the point of setting it if it can only be set to one value? Shouldn't it just be read-only?

Reply:
ebuilds that inherit `kernel-install.eclass` but not `kernel-build.eclass` will need to set it, otherwise it falls back to its current value of `${PV}${KV_LOCALVERSION}`

Reply:
Please correct me if I misunderstand, but for all the ebuilds we have in the tree currently, this:
```
if [[ -z ${KV_FULL} ]]; then
		KV_FULL=$(<"${relfile}") || die
fi
```
and this:
```
if [[ -z ${KV_FULL} ]]; then
		KV_FULL=${PV}${KV_LOCALVERSION}
fi
```
are equal right?

If I understand correctly, the point of these patches is to correct a problem where the `relfile` (what the kernel thinks the version is) is different then `${PV}${KV_LOCALVERSION}` (what portage thinks the version is and where the modules are installed). The problem occurs when the PV contains _, which is reflected in the KV as -.

But given that this discrepancy does not exist for any ebuild we already have in the tree (otherwise they wouldn't work), I don't understand the need for the "fallback path" and setting `KV_FULL` in the ebuild. Can't we simply read all versions we use from the `relfile`, both in `kernel-build.eclass` and `kernel-install.eclass`? This will not change behaviour for anything we have in the tree since for all those ebuilds it is equal to  `${PV}${KV_LOCALVERSION}`. 

Your patch looks fine to me, but it is just that it feels to me like it is also adding a knob to the eclass that it does not need.

Reply:
This fallback in particular is trying to prevent build failures caused by ebuilds that inherit `kernel-install.eclass` without setting `KV_FULL` _and_ do not have a `${WORKDIR}/build/include/config/kernel.release` file to get it from. `sys-kernel/gentoo-kernel-bin` is like this as it has its own internal variables that it uses, and falling back to `${PV}${KV_LOCALVERSION}` in `kernel-install.eclass` ensures that these ebuilds still work properly without needing a revbump. Ideally, we would revise the next `gentoo-kernel-bin` ebuild to set `KV_FULL` which means we can then get rid of this.

Reply:
Right, I see your point, in the `src_*` phases the `WORKDIR` contents of gentoo-kernel versus gentoo-kernel-bin are different.

However, `kernel-install.eclass` only really does stuff in the `pkg_preinst, pkg_postinst, pkg_config, pkg_postrm` phases. For those phases could we not read the `kernel.release` from `${ED}` (for preinst and postinst) or from `${ROOT}` (for config and postrm)?

Reply:
That leads to a chicken-and-egg scenario though. We would need to know what `KV_FULL` is so we can read `${ED}/usr/src/linux-${KV_FULL}/include/config/kernel.release` to set `KV_FULL`...

Reply:
But there is only one kernel in a package, so we can simply `find` `kernel.release` in `ED`, and export the contents as `KV_FULL`.

In any case, it's not that I think this is a huge issue. It just seems a bit unfortunate to introduce a variable that can (and sometimes must) be set, but it can only be set to one particular value.

Reply:
I'm open to implementing other ways to fix this issue at hand if you think they'd be a better fit. I like this solution simply because it allows us not only to fix the issue of underscores in the version name, but also to eliminate a whole bunch of duplication across ebuilds and eclasses, where we were sourcing `kernel.release` multiple times in multiple places... and then not even obeying it anyway.